src/Compilers/CSharp/Portable/Compiler/ClsComplianceChecker.cs

```diff
@@ -1012,7 +1012,10 @@ private bool IsCompliantType(NamedTypeSymbol type, NamedTypeSymbol context)
         private static bool IsInaccessibleBecauseOfConstruction(NamedTypeSymbol type, NamedTypeSymbol context)
         {
             // NOTE: Dev11 (incorrectly) only checks whether "type" is protected - it ignores container accessibility.
-            bool sawProtected = type.DeclaredAccessibility == Accessibility.Protected || type.DeclaredAccessibility == Accessibility.ProtectedOrInternal;
+            bool sawProtected =
+                type.DeclaredAccessibility == Accessibility.Protected ||
+                type.DeclaredAccessibility == Accessibility.ProtectedAndInternal ||
+                type.DeclaredAccessibility == Accessibility.ProtectedOrInternal;
```

Comment:
It might be better to extract this into a method taking Accessibility, there is another use site below.